assembly/src/parsers/ast/context.rs

```diff
@@ -0,0 +1,282 @@
+use super::{
+    nodes::Procedure, parse_op_token, AssemblyError, Instruction, Node, ProcMap, Token,
+    TokenStream, MODULE_PATH_DELIM,
+};
+use vm_core::utils::{
+    collections::{BTreeMap, Vec},
+    string::{String, ToString},
+};
+
+// Context
+// ================================================================================================
+
+// AST Parser context that holds internal state to generate correct ASTs.
+pub struct ParserContext {
```

Comment:
nit: I'd add `#[derive(Default)]` here. Then, you can instantiate the struct simply as `ParserContext::default()`.

Reply:
fixed